src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1447 @@
+            past_key_value=past_key_value,
+            text_end_positions=text_end_positions,
+            output_hidden_states=output_hidden_states,
+            output_attentions=output_attentions,
+            return_dict=return_dict,
+        )
+
+        sequence_output = encoder_outputs[0]
+        pooled_output = self.pooler(sequence_output) if self.pooler is not None else None
+
+        if not return_dict:
+            return (sequence_output, pooled_output) + encoder_outputs[1:]
+
+        return BaseModelOutputWithPooling(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+
+@add_start_docstrings(
+    """Beit3ForVisualReasoning has a MLP head on top of Beit3Model. Beit3 is a multimodal foundation model. The key idea
+     in BEiT-3 is to model images as another language. Beit3 uses multiway Transformers architecture which uses a
+     shared self-attention module.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3ForVisualReasoning(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super(Beit3ForVisualReasoning, self).__init__(config)
+        self.beit3 = Beit3Model(config)
+        self.classifier = Beit3MLP(config)
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(BEIT3_FOR_VISUAL_REASONING_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=SequenceClassifierOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+            self,
+            input_ids,
+            pixel_values,
+            attention_mask,
+            output_hidden_states=None,
+            output_attentions=None,
+            return_dict=None,
+            labels=None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3Processor, Beit3ForVisualReasoning
+        >>> from PIL import Image
+        >>> import requests
+        >>> import torch
+
+        >>> model = Beit3ForVisualReasoning.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> text = "This is photo of a cat"
+
+        >>> processor = Beit3Processor.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+        >>> inputs = processor(text=text, images=image, return_tensors="pt")
+
+        >>> pixel_values = torch.cat(
+        ...     (torch.tensor(inputs["pixel_values"]).unsqueeze(1), torch.tensor(inputs["pixel_values"]).unsqueeze(1)),
+        ...     dim=1,
+        ... )
+
+        >>> output = model(
+        ...     input_ids=torch.tensor(inputs["input_ids"]),
+        ...     pixel_values=pixel_values,
+        ...     attention_mask=torch.ones(inputs["input_ids"].shape),
+        ... )
+        >>> list(output.logits.shape)
```

Comment:
Same comment as below, there's no need to specify torch.tensor in case you already specify return_tensors="pt" to the processor. In this case, you can use inputs.input_ids directly.

Also this model performs natural language visual reasoning, but the example code snippet does not show how to do that. NLVR is a task of, given 2 images and one sentence, classify which of the 2 images belongs to the sentence. Would be great to illustrate that here.